### DISCUSSION TITLE
not sure how I can "setState"

### DISCUSSION THREAD

**Discussion by Ofer-Gal** [pmndrs/zustand#735] [Q&A] [answered]
In the store I have 
```
interface GridState {
   columnsState: ColumnState[];
 }
export const useGridStore = create<GridState>(persist(
    (set, get) => ({
        columnsState: []
    }),
    {
        name: "status-grid-storage", // name of item in the storage (must be unique)
        getStorage: () => localStorage, // (optional) by default the 'localStorage' is used
    }
))
```

then in the component setter I have:
```
useGridStore.setState({ columnsState: localColumnsState });
```
Does it really populate `state.columnsState` with the value of localColumnsState ?

Should I create a method in the store to populate the property? How would I do it with a parameter?

Thank you.

**Comment by dai-shi** [comment, marked as answer]
> Does it really populate `state.columnsState` with the value of localColumnsState ?

I think it does. Does it not?

> Should I create a method in the store to populate the property?

It's up to you. Both styles work: https://github.com/pmndrs/zustand/wiki/Practice-with-no-store-actions

> How would I do it with a parameter?

You can create a function or a method to take a parameter.
```js
export const setColumnState = (columnState) => {
  useGridStore.setState({ columnsState })
}
```
Or, you mean something else.

**Reply by Ofer-Gal** [reply]
Thank you. easier then expected
